roles/tests/test_user_models.py

```diff
@@ -51,3 +51,11 @@ def __init__(self, foo_var):
 
         user_group2 = FooObject('bar')
         self.assertFalse(user_group1.__eq__(user_group2))
+
+    def test_user_group_hash(self):
+        """Test for UserGroup __hash__"""
+        user_group = UserGroup('foo1', 'bar')
+        self.assertEqual(
+            user_group.__hash__(),
+            hash((user_group.username, user_group.group_type))
```

Comment:
I don't think this is testing Python's `hash()` function. It tests that when you `hash()` a `UserGroup`, you're definitely hashing on the two fields you think you are.

Even if you decide not to instantiate a second `UserGroup`, which isn't strictly necessary, I definitely think there should be tests for changing the two fields (one at a time) and proving that makes the hashes different.